### PR TITLE
Feature/redmine4.0

### PR DESCRIPTION
**Pull request by dmartingit**

### CODE REVIEW

**Comment by dmartingit** [comment]
After manually testing everything. Some things are not working as expected which the tests did not show.

**Comment by dmartingit** [comment]
An exception was raised in our form fields generation because false values passed to include_blank in a `select` would be ignored if the field was required: rails/rails#20124

File: app/views/hourglass_ui/forms/fields/_activity.slim [hunk 1]
```diff
@@ -2,4 +2,4 @@
   .label
     = form.label :activity_id
   .input
-    = form.collection_select :activity_id, TimeEntryActivity.applicable(entry.project), :id, :name, {include_blank: !local_assigns[:required]}, disabled: local_assigns[:disabled], required: local_assigns[:required]
+    = form.collection_select :activity_id, TimeEntryActivity.applicable(entry.project), :id, :name, { include_blank: local_assigns[:required] }, disabled: local_assigns[:disabled], required: local_assigns[:required]
```
[the remark this answers is not in the document]
Comment (arBmind) [reply app/views/hourglass_ui/forms/fields/_activity.slim]: Does this still work with the old Redmine / Rails versions?

[the remark this answers is not in the document]
Comment (dmartingit) [reply app/views/hourglass_ui/forms/fields/_activity.slim]: We have the following two cases: 

1. Select tag is not required -> include a blank field
2. Select tag is required -> no blank field

What has changed is that we can no longer have a required select tag without a blank option.
The reason is that the HTML5 standard requires that behavior and rails 5 adopted it.
This leads to a changing behavior in the 2. case and the closest behavior we can create is imo to always include a blank field in a select tag.

**Pull request merged**